### PR TITLE
Support koa2, and upgrade dependencies

### PR DESCRIPTION
**Pull request by halfcrazy**
koa2 no longer encourages to use generator functions, to replace generator function with async function so that dont have to use co to workaround with koa2.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -67,16 +68,17 @@ exports = module.exports = function (app, settings) {
    * @param {Object} options
    * @return {String} html
    */
-  function *render(view, options) {
+  function render(view, options) {
```
Comment (dead-horse) [inline index.js]: please use async function to implement.

Comment (halfcrazy) [inline index.js]: Of course, working in progress.

**Comment by halfcrazy** [comment]
@dead-horse Changed to async/await syntax.

File: index.js [hunk 2]
```diff
@@ -10,16 +10,18 @@
  * Module dependencies.
  */
 
-var copy = require('copy-to');
-var path = require('path');
-var fs = require('co-fs');
-var ejs = require('ejs');
+const Promise = require('bluebird');
+const debug = require('debug')('koa-ejs');
+const readFile = Promise.promisify(require("fs").readFile);
```
Comment (dead-horse) [inline index.js]:
```js
const fs = require('mz').fs;

await fs.readFile
```

File: index.js [hunk 3]
```diff
@@ -10,16 +10,18 @@
  * Module dependencies.
  */
 
-var copy = require('copy-to');
-var path = require('path');
-var fs = require('co-fs');
-var ejs = require('ejs');
+const Promise = require('bluebird');
```
Comment (dead-horse) [inline index.js]: remove require of bluebird

File: index.js [hunk 4]
```diff
@@ -90,29 +93,36 @@ exports = module.exports = function (app, settings) {
...
-      //normal operation
-      this.type = 'html';
-      this.body = html;
-    }else{
-      //only return the html
-      return html;
-    }
+  app.context.render = async function (view, _context) {
+    const ctx = this;
+    return new Promise(async (resolve, reject) => {
```
Comment (dead-horse) [inline index.js]: no need to use new Promise here

Comment (dead-horse) [inline index.js]: write with async function is just like write code with co, no need to change the logic.

Comment (halfcrazy) [inline index.js]: @dead-horse ping

File: index.js [hunk 5]
```diff
@@ -33,7 +34,7 @@ var defaultSettings = {
  *
  * usage:
  * ```
- * yield *this.render('user', {name: 'dead_horse'});
+ * ctx.render('user', {name: 'dead_horse'});
```
Comment (dead-horse) [inline index.js]: await ctx.render

File: package.json [hunk 6]
```diff
@@ -26,22 +26,17 @@
...
-    "copy-to": "~2.0.1",
-    "ejs": "^2.3.4"
+    "debug": "^2.6.1",
+    "ejs": "^2.5.6",
+    "mz": "^2.6.0"
   },
   "engine": {
-    "node": ">0.11"
-  },
-  "files": [
```
Comment (dead-horse) [inline package.json]: keep `files`

**Comment by dead-horse** [comment]
👍

**Comment by dead-horse** [comment]
4.0.0

**Pull request merged**